### PR TITLE
Fix npm use webpack

### PR DESCRIPTION
**Pull request by iam4x**
Hello @RSO @martijnrusschen 

This a big PR to make the lib compatible with module loaders, like in CommonJS in node. (Should fix https://github.com/Hacker0x01/react-datepicker/issues/88)

#### Notables changes

* Builded files are now in `/dist` directory (updated paths references in `package.json` and `bower.json`)
* Added minified builds too (`react-datepicker.min.css` and `react-datepicker.min.js`)
* Changed the whole build process with **webpack** (http://webpack.github.io/)

#### All changes

* Use of webpack instead of browserify (http://webpack.github.io/)
* Use of `babel-jest` for preprocessor into tests
* Use of `babel-loader` for JSX preprocessor into build
* Remove browserify, reactify, browserify-shim...
* Remove `/** @jsx React.DOM **/` since it's deprecated (dups of PR https://github.com/Hacker0x01/react-datepicker/pull/86)
* Fix global var leak (dups of PR https://github.com/Hacker0x01/react-datepicker/pull/87)
* Add `$ grunt webpack:(min | unmin | example)` for JSX building files
* Add `$ grunt sass:(min | unmin)` for building styles files

#### Next?

Remove `Tether` dep to make the lib isomorphic, at this state we can't run it on the server.

Best regards, Maxime.

### CODE REVIEW

**Comment by RSO** [comment]
Thanks for this, it's a lot to take in at once, but I hope to do a first run through the PR's today, and get some real work done in this repository tomorrow!

**Comment by RSO** [comment]
Could you elaborate as to why we need webpack in favour of browserify to solve this issue? (a link/whatever would be fine as well)

**Comment by iam4x** [comment]
I've tried to have a same build for browser and node with browserify, couldn't achieve it. Mainly because of this:

With webpack, you can define externals namespaces for every use (amd, commonjs & browser):

```
{
      'react/addons': {
        root: 'React',
        commonjs2: 'react',
        commonjs: 'react',
        amd: 'react'
      }
    },
    {
      'moment': {
        root: 'moment',
        commonjs2: 'moment',
        commonjs: 'moment',
        amd: 'moment'
      }
    },
    {
      'react-onclickoutside': {
          root: 'OnClickOutside',
          commonjs2: 'react-onclickoutside',
          commonjs: 'react-onclickoutside',
          amd: 'react-onclickoutside'
      }
    }
```

With browserify and browserify-shim it wasn't enough, since `react/addons` and `react-onclickoutside` have different namespaces depending on environment. Actual:

```
"browserify-shim": {
    "react/addons": "global:React",
    "moment": "global:moment",
    "react-onclickoutside": "global:OnClickOutside"
  },
```

If you find something for browserify, I'm happy to give it another shot but I lost almost a day wrapping my head around this issue :)

**Comment by RSO** [comment]
Cool! Yeah, this is my first project with both browserify and webpack, so I just wasn't sure about all the pros/cons. Thanks again for taking the time to fix this, I'll check it out and see what I can do. Would you mind if I'd try to split this PR up in multiple PRs? I'm unsure if it's needed, but it might help me get an understanding of the whole situation and fixing #88 sooner.

**Comment by iam4x** [comment]
Sure,

First you can merge https://github.com/Hacker0x01/react-datepicker/pull/86 and https://github.com/Hacker0x01/react-datepicker/pull/87 which are included into my PR, I needed them.

Then I will rebase and try to breakdown things.

File: gruntfile.js [hunk 1]
```diff
@@ -29,8 +54,8 @@ module.exports = function(grunt) {
         tasks: ['sass']
       },
 
-      browserify: {
-        files: ['src/**/*.js'],
+      webpack: {
+        files: ['src/**/*.js', 'src/**/*.jsx'],
         tasks: ['browserify']
```
Comment (RSO) [inline gruntfile.js]: `browserify` -> `webpack`

**Comment by RSO** [comment]
I just tested this and everything looks good. I landed the other PR's, so you should be able to rebase this PR.

When I run `grunt build` my `react-datepicker.min.js` file changes, is it outdated, or are we running different versions of the dependencies?

Webpack seems interesting, shouldn't we also be able to drop SCSS compilation out of the gruntfile and let webpack handle it?

**Comment by FoxxMD** [comment]
@iam4x any idea when this can be merged?

**Comment by RSO** [comment]
I'll see if I can find the time to continue work on this later this week. Sorry for the delay!

**Comment by ryan1234** [comment]
No pressure, but I'd also love to see this PR merged. =)

**Comment by RSO** [comment]
Damn, dat pressure ;) I'm feeling confident that I'll have time to work on this tonight! Just sit tight!

**Comment by FoxxMD** [comment]
We :heart: you @RSO ! We're on your time.

**Comment by iam4x** [comment]
I'll have some time tomorrow to review and rebase my branch :+1:

**Comment by jmfurlott** [comment]
What is the status of this merge? As well as updating the package to work with React 0.13? Can these PRs be merged?

**Comment by RSO** [comment]
Merged through #93. Thank you so much for the help! And the patience ;-). I'll get a new release out today!

**Comment by FoxxMD** [comment]
:heart:

**Pull request closed without merging**